### ISSUE TITLE
PHP 7.2 Warnings

### ISSUE THREAD

**Issue by scofennell** [stripe/stripe-php#587]
Hi, I when I use WP-Enginge's php 7.2 Compatibility Checker (https://wordpress.org/plugins/php-compatibility-checker/), get the following warnings:

```
FILE: /nas/content/staging/lxbplatform/wp-content/plugins/lxb-account/lib/stripe/lib/StripeObject.php\
---------------------------------------------------------------------------------------------------------------------------------------------------------\
FOUND 0 ERRORS AND 2 WARNINGS AFFECTING 2 LINES\
---------------------------------------------------------------------------------------------------------------------------------------------------------\
 293 | WARNING | Method name "StripeObject::__toJSON" is discouraged; PHP has reserved all method names with a double underscore prefix for future use.\
 308 | WARNING | Method name "StripeObject::__toArray" is discouraged; PHP has reserved all method names with a double underscore prefix for future use.\
---------------------------------------------------------------------------------------------------------------------------------------------------------\
```

**Comment by ob-stripe** [comment]
Hi @scofennell, thanks for the report.

This is a dupe of #558, but I think maybe we should rename the methods -- neither `__toJSON` nor `__toArray` are real magic methods, we just reused the naming convention.

This would be a breaking change though since some users may be relying on these methods in their integrations, so it would have to be done in the next major version. I'll tag this issue as `future` so that it stays on our radar.

**Comment by ob-stripe** [comment]
This is fixed in stripe-php 7.0.0. `__toJSON` has been renamed to `toJSON` and `__toArray` has been renamed to `toArray`.

Additionally, `toArray` no longer takes any parameters -- the conversion is always done recursively.
